From 215663dbf3f24f729b87eea090b3f84dd679d509 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Fri, 13 Jun 2014 18:30:23 +0200
Subject: [PATCH 22/30] MINOR: config: warn when tcp-check rules are used
 without option tcp-check

Since this case means that the rules will be ignored, better emit a warning.
---
 src/cfgparse.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/cfgparse.c b/src/cfgparse.c
index 00b7582..017bcae 100644
--- a/src/cfgparse.c
+++ b/src/cfgparse.c
@@ -6504,6 +6504,13 @@ out_uri_auth_compat:
 			memcpy(curproxy->check_req, sslv3_client_hello_pkt, curproxy->check_len);
 		}
 
+		if (!LIST_ISEMPTY(&curproxy->tcpcheck_rules) &&
+		    (curproxy->options2 & PR_O2_CHK_ANY) != PR_O2_TCPCHK_CHK) {
+			Warning("config : %s '%s' uses tcp-check rules without 'option tcp-check', so the rules are ignored.\n",
+				proxy_type_str(curproxy), curproxy->id);
+			err_code |= ERR_WARN;
+		}
+
 		/* ensure that cookie capture length is not too large */
 		if (curproxy->capture_len >= global.tune.cookie_len) {
 			Warning("config : truncating capture length to %d bytes for %s '%s'.\n",
-- 
1.8.3.2

